packages/core/src/credential/Credential.ts

```diff
@@ -332,6 +339,43 @@ export async function verifyPresentation(
   })
 }
 
+/**
+ * Queries the attestation record for a credential and matches their data. Fails if no attestation exists, if it is revoked, or if the attester is unknown.
+ *
+ * @param credential The [[ICredential]] whose attestation status should be checked.
+ * @param trustedAttesters A list of one or more attesters who are accepted as issuers of the credential. If the actual attester is not in trustedAttesters, and this is not a delegated attestation, verification fails.
+ * @param trustedDelegators An optional list of accepted delegators. If this is a delegated attestation and its issuer is not trusted, verification will still be successful if any of these identities is the owner of an ancestor node in the delegation tree.
+ */
+export async function verifyAttestation(
+  credential: ICredential,
+  trustedAttesters: DidUri[],
+  trustedDelegators: DidUri[] = []
+): Promise<void> {
+  const api = ConfigService.get('api')
+  const { rootHash } = credential
+  const maybeAttestation = await api.query.attestation.attestations(rootHash)
+  if (maybeAttestation.isNone)
+    throw new SDKErrors.CredentialUnverifiableError('Attestation not found')
+  const attestation = Attestation.fromChain(maybeAttestation, rootHash)
+  if (
+    credential.claim.cTypeHash !== attestation.cTypeHash ||
+    credential.delegationId !== attestation.delegationId
+  ) {
+    throw new SDKErrors.CredentialUnverifiableError(
+      'Attestation does not match credential'
+    )
+  }
+  if (attestation.revoked)
```

Comment:
BTW I always prefer to add `{}`. This reduces the chance of potential issues where you would want to have something in the if but it's only indented and not really inside. There was a huge security issue because of this once (i think it was in some apple product). They also didn't use `{}` for ifs and because of a merge error a `goto verified` was added twice. The second one was always executed even though the check wasn't valid. -> security issue. 😨

Reply:
uuuu damn that's nasty. will be adding {} everywhere

Reply:
https://github.com/KILTprotocol/sdk-js/pull/702/commits/0b7a7e468f206cda0720c5eebbf8cc26e0b8b78a